### PR TITLE
Add pure function component snippet

### PR DESCRIPTION
**Pull request by mlienau**
As described here, https://facebook.github.io/react/docs/components-and-props.html, pure components can be defined with a plain Javascript function.  This PR adds a snippet for creating a pure function component.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vscode-react-typescript",
-  "version": "1.0.4",
+  "version": "1.0.5",
```
Comment (infeng) [inline package.json]: you dont`t  need to modify the version

File: snippets/snippets.json [hunk 2]
```diff
@@ -83,6 +83,23 @@
 		],
 		"description": "Create a React PureComponent without import and export."
 	},
+	"React Pure Function Component": {
+		"prefix": "tsrpfc",
+		"body": [
+			"export interface ${1:App}Props {",
+			"}",
+			"",			
+			"export function ${1:} (props: ${1:}Props) {",
```
Comment (infeng) [inline snippets/snippets.json]: I think ComponentName should be `${0}`

Comment (mlienau) [inline snippets/snippets.json]: According to the [docs](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_tabstops) `{0}` represents the final cursor position, so in this case you would enter the component name, hit enter and you'd wind up in the render section.  Though I'm thinking now we should end inside props instead.

File: package.json (hunk 1 above)
Comment (mlienau) [inline package.json]: Doesn't the version number need to be bumped and the extension re-packaged in order to show as an update in the extensions tab?

Comment (infeng) [inline package.json]: A new version is not only on change, let me change the version number and create a tag

File: snippets/snippets.json (hunk 2 above)
Comment (infeng) [inline snippets/snippets.json]: you are right.

File: snippets/snippets.json [hunk 3]
```diff
@@ -83,6 +83,25 @@
...
 	},
+	"React Pure Function Component": {
+		"prefix": "tsrpfc",
+		"body": [
+			"import * as React from 'react';",
+			"",
+			"export interface ${1:App}Props {",
+			"}",
+			"",			
+			"export function ${1:} (props: ${1:}Props) {",
```
Comment (infeng) [inline snippets/snippets.json]: `export` should be `export default` better.

**Pull request merged**